schemas/pa-yaml/v2.2/pa.schema.yaml

```diff
@@ -0,0 +1,350 @@
+# Unfortunately, it also seems to make it so that schema cache doesn't get updated if the schema file updates.
+# WORKAROUND: Close the VSCode window and reopen.
+
+$schema: http://json-schema.org/draft-07/schema#
+$id: http://powerapps.com/schemas/pa-yaml/v2.2/pa.schema
+title: Microsoft Power Apps schema for app source yaml files (v2.2).
+description: >-
+  The schema for all *.pa.yaml files which are used to describe a Power Apps canvas app.
+  All *.pa.yaml files in an *.msapp are logically combined into a single *.pa.yaml file.
+
+type: object
+additionalProperties: false
+properties:
+  App:
+    $ref: "#/definitions/App-instance"
+  Screens:
+    $ref: "#/definitions/Screens-name-instance-map"
+  ComponentDefinitions:
+    $ref: "#/definitions/ComponentDefinitions-name-instance-map"
+
+definitions:
+  App-instance:
+    description: Represents the top-level App entity.
+    type: object
+    additionalProperties: false
+    properties:
+      Properties: { $ref: "#/definitions/Properties-formula-map" }
+
+      # Note: App children is fixed.
+      Children:
+        description: The App currently only supports the 'Host' child.
+        type: object
+        additionalProperties: false
+        properties:
+          Host:
+            type: object
+            additionalProperties: false
+            properties:
+              # Currently, the Control identifier is static, but may need to be exposed in order to support variants
+              #Control: { $ref: "#/definitions/Control-type-identifier" }
+              Properties: { $ref: "#/definitions/Properties-formula-map" }
+
+  Screens-name-instance-map:
+    type: object
+    propertyNames: { $ref: "#/definitions/Screen-name" }
+    additionalProperties:
+      type: object
+      additionalProperties: false
+      properties:
+        Properties: { $ref: "#/definitions/Properties-formula-map" }
+        Children: { $ref: "#/definitions/Children-Control-instance-sequence" }
+
+  Screen-name:
+    description: The base requirements for a named entity in an app.
+    type: string
+    pattern: "^[a-zA-Z0-9_- ]+$"
```

Comment:
I think screen name can have more special characters

Reply:
sure, I'll comment out the patterns for these for now until we know what the actual char set that's allowed.